src/app/component/Slider/Slider.component.js

```diff
@@ -46,12 +68,50 @@ class Slider extends Component {
         }, 300);
     }
 
-    getSlideSize() {
-        const { isVertical } = this.props;
+    componentDidUpdate(prevProps, prevState) {
+        const { activeImage: prevActiveImage } = prevProps;
+        const { activeImage, isVertical } = this.props;
+        const { activeSlide: prevActiveSlide } = prevState;
+        const { activeSlide } = this.state;
 
-        return isVertical
-            ? this.sliderRef.current.offsetHeight
-            : this.draggableRef.current.offsetWidth;
+        if (activeImage !== prevActiveImage || prevActiveSlide !== activeSlide) {
+            const newTranslate = activeSlide * this.getSlideSize();
+
+            CSS.setVariable(
+                this.draggableRef,
+                'animation-speed',
+                `${ Math.abs((prevActiveSlide - activeSlide) * 300) }ms`
+            );
+
+            CSS.setVariable(
+                this.draggableRef,
+                isVertical ? 'translateY' : 'translateX',
+                `${newTranslate}px`
+            );
+        }
+    }
+
+    onClickChangeSlide(state, slideSize, lastTranslate, fullSliderSize) {
+        const { originalX, prevActiveSlider } = state;
```

Comment:
This code does not support the vertical slide. Maybe we should remove the vertical option for slider completely? The logic seem unused throughout the website.